### ISSUE TITLE
Querying data fails when Enhanced Domains is enabled

### ISSUE THREAD

**Issue by svenfinitiv** [forcedotcom/cli#1649]
Issue also occurs when retrieving, but query was easier to provide steps to reproduce.

sfdx force:data:soql:query -u dev -q "SELECT Id FROM Account LIMIT 1"

Querying Data... done
ERROR running force:data:soql:query:  Cannot read properties of undefined (reading 'length')
Type: Cannot read properties of undefined (reading 'length')
    at Query._execute (/Users/.../.local/share/sfdx/client/7.161.0-7a9b98c/node_modules/@salesforce/core/node_modules/jsforce/lib/query.js:482:205)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /Users/.../.local/share/sfdx/client/7.161.0-7a9b98c/node_modules/@salesforce/core/node_modules/jsforce/lib/query.js:394:9

The same query works when running against a sandbox without enhanced domains enabled:

sfdx force:data:soql:query -u qs -q "SELECT Id FROM Account LIMIT 1"

 ID
 ──────────────────
 0011j00001Ee7RFAAZ
Total number of records retrieved: 1.
Querying Data... done

sfdx version:

 CLI Version :
	sfdx-cli/7.161.0

 Architecture:
	darwin-x64

 Node Version :
	node-v16.16.0

 Plugin Version:
	@oclif/plugin-autocomplete 0.3.0 (core)
	@oclif/plugin-commands 1.3.0 (core)
	@oclif/plugin-help 3.3.1 (core)
	@oclif/plugin-not-found 1.2.6 (core)
	@oclif/plugin-plugins 1.10.11 (core)
	@oclif/plugin-update 1.5.0 (core)
	@oclif/plugin-warn-if-update-available 1.7.3 (core)
	@oclif/plugin-which 1.0.4 (core)
	@salesforce/sfdx-plugin-lwc-test 1.0.0 (core)
	alias 2.1.0 (core)
	apex 1.1.0 (core)
	auth 2.2.3 (core)
	community 2.0.0 (core)
	config 1.4.14 (core)
	custom-metadata 2.0.0 (core)
	data 2.0.4 (core)
	generator 2.0.2 (core)
	info 2.0.1 (core)
	limits 2.0.1 (core)
	org 2.0.5 (core)
	salesforce-alm 54.6.3 (core)
	schema 2.1.1 (core)
	sfdx-cli 7.161.0 (core)
	signups 1.2.0 (core)
	source 2.0.9 (core)
	telemetry 2.0.0 (core)
	templates 55.0.0 (core)
	trust 2.0.1 (core)
	user 2.1.0 (core)

 OS and Version:
	Darwin 21.5.0

**Comment by peternhale** [comment]
@svenfinitiv I created a new scratch org, which has enhanced domains enabled by default, and ran your example query. The query was successful.

```
sfdx force:data:soql:query -u test-ldhxfjzme5k2@example.com -q "SELECT Id FROM Account LIMIT 1"
 ID                 
 ────────────────── 
 00121000010hjEMAAY 
Total number of records retrieved: 1.
```

Anything special about your scratch org config that might be different than mine?

```
{
    "orgName": "Dreamhouse",
    "edition": "Developer",
    "features": ["Walkthroughs", "EnableSetPasswordInApi"],
    "settings": {
        "lightningExperienceSettings": {
            "enableS1DesktopEnabled": true
        },
        "mobileSettings": {
            "enableS1EncryptedStoragePref2": false
        }
    }
}
```

**Comment by svenfinitiv** [comment]
SOLVED: sfdx auth:logout / sfdx auth:web:login for the affected org did the trick.

Hi @peternhale,

it's a DevPro sandbox and the only difference really is the Enhanced Domains setting. I rechecked the issue some minutes ago, then disabled the Enhanced Domains setting and rerun the query - now with the expected result. To revalidate I reenabled the ED only to run into the issue again.

To follow the usual "turn it off and on again" I reinstalled sfdx without success, but logging out and in again (auth:logout, auth:web:login) fixed the issue.

I'm closing this issue with the new knowledge, thanks for taking your time to check and I hope if someone else stumbles over this issue they can more easily solve theirs.

Cheers, Sven
